src/core/repo.rs

```diff
@@ -26,6 +28,20 @@ impl Target for Repo {
     fn template_context(&self, config: &Config) -> Value {
         repo_context(config, self)
     }
+
+    fn rename(&mut self, new_name: &str) -> std::io::Result<()> {
+        let parent = self.path.parent().ok_or_else(|| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                format!("Cannot determine parent of path: {}", self.path.display()),
+            )
+        })?;
+
+        let old_path = &self.path.clone();
+        self.path = parent.join(new_name);
```

Comment:
I wonder if we want to support renaming across repo namespaces (i.e. moving repos from one org to another) in addition to just changing the name of the repo itself.

Something like:

```bash
gt rename gh:SierraSoftworks/git-tool gh:notheotherben/git-tool
```

If so, we'd presumably want this to take a `src: &Repo, dst: &Repo` to describe where to move things (this approach also allows us to get clever about hooking move behaviour in the service logic so that we can call GitHub APIs to orchestrate a remove on the remote).

Reply:
Funny that you mention this. Because this is the exact thing that sparked this effort.
I wanted to rename `gh:specht-labs/*` to `gh:SpechtLabs/*` but then I started with _just_ renaming the repositories as this seemed to be easier and would allow me to re-use some of the same functionality